flyteplugins/go/tasks/pluginmachinery/flytek8s/non_interruptible.go

```diff
@@ -0,0 +1,92 @@
+package flytek8s
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/flyteorg/flyte/flyteidl/gen/pb-go/flyteidl/core"
+	pluginsCore "github.com/flyteorg/flyte/flyteplugins/go/tasks/pluginmachinery/core"
+)
+
+// Wraps a regular TaskExecutionMetadata and overrides the IsInterruptible method to always return false
+// This is useful as the runner and the scheduler pods should never be interruptible
+type NonInterruptibleTaskExecutionMetadata struct {
+	metadata pluginsCore.TaskExecutionMetadata
+}
+
```

Comment:
Is there a reason that we're using this as a separate field rather than composition? Then we can just mask the `IsInterruptible` instead of everything?
```
type NonInterruptibleTaskExecutionMetadata struct {
    pluginsCore.TaskExecutionMetadata
}

func (n NonInterruptibleTaskExecutionMetadata) IsInterruptible() bool {
	return false
}
```